coral/src/app/features/topics/details/schema/TopicDetailsSchema.tsx

```diff
@@ -11,56 +12,99 @@ import {
+import { Link, useNavigate } from "react-router-dom";
 import { useTopicDetails } from "src/app/features/topics/details/TopicDetails";
 import { SchemaPromotionBanner } from "src/app/features/topics/details/schema/components/SchemaPromotionBanner";
 import { SchemaStats } from "src/app/features/topics/details/schema/components/SchemaStats";
 
 function TopicDetailsSchema() {
-  const { topicName, environmentId } = useTopicDetails();
+  const navigate = useNavigate();
+  const {
+    topicName,
+    topicSchemas: {
+      allSchemaVersions = [],
+      latestVersion,
+      schemaDetailsPerEnv,
+    },
+    setSchemaVersion,
+    topicOverview: { topicInfoList },
+  } = useTopicDetails();
+
+  const isTopicOwner = topicInfoList[0].topicOwner;
+  const noSchema = allSchemaVersions.length === 0;
 
   function promoteSchema() {
     console.log("dummy function");
   }
 
+  if (noSchema) {
+    return (
+      <>
+        <PageHeader title="Schema" />
+        <EmptyState
+          title="No schema available for this topic"
+          primaryAction={{
+            onClick: () => navigate(`/topic/${topicName}/request-schema`),
+            text: "Request a new schema",
+          }}
+        />
+      </>
+    );
+  }
+
   return (
     <>
       <PageHeader title="Schema" />
+
       <Box display={"flex"} justifyContent={"space-between"}>
         <Box display={"flex"} colGap={"l1"}>
           <NativeSelect
             style={{ width: "300px" }}
             aria-label={"Select version"}
+            onChange={(e) => setSchemaVersion(Number(e.target.value))}
+            defaultValue={schemaDetailsPerEnv?.version}
           >
-            <Option key={"1"} value={"1"}>
-              version 1
-            </Option>
+            {allSchemaVersions.map((version) => (
+              <Option key={version} value={version}>
+                Version {version} {version === latestVersion && "(latest)"}
+              </Option>
+            ))}
           </NativeSelect>
-          <Typography.SmallTextBold color={"grey-40"}>
+          <Typography.SmallStrong color={"grey-40"}>
             <Box display={"flex"} marginTop={"3"} colGap={"2"}>
               <Icon icon={gitNewBranch} style={{ marginTop: "2px" }} />{" "}
-              <span>5 versions</span>
+              <span>{allSchemaVersions.length} versions</span>
             </Box>
-          </Typography.SmallTextBold>
+          </Typography.SmallStrong>
         </Box>
 
-        <Box alignSelf={"top"}>
-          <Button.ExternalLink
-            //@TODO verify if environmentId is the right one
-            href={`/topic/${topicName}/request-schema?env=${environmentId}`}
-            icon={add}
-          >
-            Request a new version
-          </Button.ExternalLink>
-        </Box>
+        {isTopicOwner && (
+          <Box alignSelf={"top"}>
+            <Link
+              to={`/topic/${topicName}/request-schema?env=${schemaDetailsPerEnv?.env}`}
```

Comment:
I think we should only show the button if there is an env instead of potentially passing `undefined` in the url. Because if there is no `schemaDetailsPerEnv.env` it's basically a sign of an error at this point, right? So only show link if it's there and maybe even throw an error if it's not? In a product I would do that, so monitoring can catch up on that error and we know that there was data expected that didn't pass. We don't know if our users use Klaw with error monitoring, so it could help them debugging? (this is probably something that we can think about in general -> how to surface errors to user (not end user but devs) so they can handle it)

Reply:
I think this is API inconsistency, I'm pretty sure `env` will always be defined on `schemaDetailsPerEnv`. @aindriu-aiven , is that right?

Reply:
Probably, but as long as it _could_ be undefined without us noticing, we should treat it like that? 🤔 Or update api soonish :D

Reply:
However, `schemaDetailsPerEnv` itself might not be defined, if there is no schema for the chosen topic env :o

Reply:
These two commits address your conerns, I believe:
- [8a706d1](https://github.com/aiven/klaw/pull/1344/commits/8a706d1f9b79262a2127b00120754bcbc3681e2c): fix the typs of the properties of `schemaDetailsPerEnv` to be correctly required
- [d13dddb](https://github.com/aiven/klaw/pull/1344/commits/d13dddb4f2627906675d68837b813451decab042): `noSchema` is also `true` if `schemaDetailsPerEnv` is `undefined`. Also remove the now superfluous optional chaining.

Reply:
noice!